pkg/controller/statefulset/stateful_set_control_test.go

```diff
@@ -208,7 +211,20 @@ func TestStatefulSetControl(t *testing.T) {
 	}
 }
 
-func CreatesPods(t *testing.T, set *apps.StatefulSet, invariants invariantFunc) {
+func CreatesPodsWithPodIndexLabelFeature(t *testing.T, set *apps.StatefulSet, invariants invariantFunc) {
+	createPods(t, set, invariants, true)
+}
+
+func CreatesPodsWithoutPodIndexLabelFeature(t *testing.T, set *apps.StatefulSet, invariants invariantFunc) {
+	createPods(t, set, invariants, false)
+}
+
+func createPods(t *testing.T, set *apps.StatefulSet, invariants invariantFunc, isPodIndexLabelEnabled bool) {
+	if !isPodIndexLabelEnabled {
+		// TODO: this will be removed in 1.35
+		featuregatetesting.SetFeatureGateEmulationVersionDuringTest(t, feature.DefaultFeatureGate, utilversion.MustParse("1.31"))
+		featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.PodIndexLabel, false)
+	}
```

Comment:
I believe:
```golang
if !isPodIndexLabelEnabled {
		// TODO: this will be removed in 1.35
		featuregatetesting.SetFeatureGateEmulationVersionDuringTest(t, feature.DefaultFeatureGate, utilversion.MustParse("1.31"))
}
featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.PodIndexLabel, isPodIndexLabelEnabled)
```
will read better